### PR TITLE
Fixes #17527 - Add support for hostgroup facets

### PR DESCRIPTION
**Pull request by ShimShtein**
This is the first commit. The facet is already functioning.
Things that are missing:

- [x] support for inheritance
- [x] support for same facet in both host and hostgroup

### CODE REVIEW

**Comment by ShimShtein** [comment]
Finally, the tests are passing

**Comment by mmoll** [comment]
this needs a rebase :)

**Comment by ShimShtein** [comment]
I have rebased this PR.

File: app/models/concerns/facets/hostgroup_extensions.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+module Facets
+  module HostgroupExtensions
+    extend ActiveSupport::Concern
+    include Facets::ModelExtensionsBase
+
+    included do
+      configure_facet(:hostgroup, :hostgroup, :hostgroup_id) do |facet_config|
+        # add_ancestry_class_translation(facet_config.name.to_s => facet_config.hostgroup_configuration.model.name)
+        # nested_attribute_for "#{facet_config.name}_id"
```
Comment (mmoll) [inline app/models/concerns/facets/hostgroup_extensions.rb]: what's that? ;)

**Comment by mmoll** [comment]
@ShimShtein could you prepare a corresponding Katello PR?

**Comment by mmoll** [comment]
@ShimShtein that rebase went wrong, I guess?

**Comment by ShimShtein** [comment]
@mmoll I am adding a couple of fixes to hostgroup part, that were needed for Katello hostgroup PR.
I will squash all the commits once both PRs would be ready.

**Comment by mmoll** [comment]
@ShimShtein please check the test run output, there's something wrong, IMHO... 

Also, please rebase to get Rails 5.2 :)

**Comment by mmoll** [comment]
@ShimShtein I rebased this onto current master to fix the test failures and everything came back green. Now a rebase is needed, because GH-5837 got merged.

**Comment by ShimShtein** [comment]
Rebased

**Comment by ShimShtein** [comment]
[test foreman]

Not sure that the failure is related (it's on host <-> location association and only for specific ruby version)

**Comment by mmoll** [comment]
:green_heart:

**Review by mmoll — DISMISSED** [review]
If there are no further objections, I'm merging this later today before the 1.20 quiet period starts.

**Comment by tbrisker** [comment]
So we don't get to the 1.21 stabilization period with this - can you please rebase @ShimShtein  one last time just to verify this still works with the latest code? if it passes i'm fine with merging it now so we can fix any issues if needed in time for 1.21.

**Review by tbrisker — APPROVED** [review]
Looks like tests are still passing. Merging now so we have time to fix any issues before 1.21.

**Comment by tbrisker** [comment]
Just 2 years to merge! never despair! :smile:

**Pull request merged**